android/libraries/rib-base/src/main/java/com/badoo/ribs/android/PermissionRequesterImpl.kt

```diff
@@ -0,0 +1,157 @@
+package com.badoo.ribs.android
+
+import android.annotation.TargetApi
+import android.content.pm.PackageManager
+import android.os.Build
+import android.support.v4.app.ActivityCompat
+import android.support.v4.content.ContextCompat
+import android.support.v7.app.AppCompatActivity
+import com.badoo.ribs.android.PermissionRequester.CheckPermissionsResult
+import com.badoo.ribs.android.PermissionRequester.RequestPermissionsEvent
+import com.badoo.ribs.android.PermissionRequester.RequestPermissionsEvent.Cancelled
+import com.badoo.ribs.android.PermissionRequester.RequestPermissionsEvent.RequestPermissionsResult
+import com.badoo.ribs.core.Identifiable
+import com.badoo.ribs.core.requestcode.RequestCodeRegistry
+import com.jakewharton.rxrelay2.PublishRelay
+import io.reactivex.Observable
+
+class PermissionRequesterImpl(
+    private val activity: AppCompatActivity,
+    private val requestCodeRegistry: RequestCodeRegistry
+) : PermissionRequester, PermissionRequestResultHandler {
+
+    private val events = HashMap<Int, HashMap<Int, PublishRelay<RequestPermissionsEvent>>>()
```

Comment:
I would suggest to have one relay with filter instead of hashmap of hashmaps.
Then we can just wrap the RequestPermissionEvent with data class holding it, requestCode and groupId. I feel like this should be a bit simpler than this construct.

Reply:
I'm waiting for everyone's opinion on this. So far there's 1 for, 1 against the current impl, and I don't have strong feelings about it.

Reply:
I like the current approach with streams.
From the user perspective it's easier, looks nice. From our side: map of map, yes, it's not obvious. Let's try to keep API and make this map easier to understand.

Naming, comment, separate meaningful methods for operating over the map.

Reply:
Going with single stream approach after group discussion